src/ASM/ASMs2Dmx.C

```diff
@@ -184,8 +185,9 @@ bool ASMs2Dmx::generateFEMTopology ()
                                             ASMmxBase::FULL_CONT_RAISE_BASIS1).front();
     else
       projBasis = m_basis.front();
+    delete surf;
   }
-  delete surf;
```

Comment:
When the grid is shared, you can't delete `surf`, because then the object pointed to already has been deleted by the owner patch, that's why valgrind give you a message on this line, I think. It is still being deleted when the grid _is_ shared, that's why its is moved inside the `if (m_basis.empty())` block. But it can be the fix has to be different. I only noticed that valgrind stopped complaining about it (did not check for leakage).